### PR TITLE
UIREQ-944 create Jest/RTL test for NoteEditRoute.js

### PR DESCRIPTION
**Pull request by cchitneedi**
Refs: 
UIREQ-944 create Jest/RTL test for NoteEditRoute.js
URL: https://issues.folio.org/browse/UIREQ-944

### CODE REVIEW

File: src/routes/NoteEditRoute.test.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+const renderNoteEditRoute = (data1, data2) => render(
+  <Router history={history}>
+    <NoteEditRoute location={data1} history={data2} match={match} />
+  </Router>
+);
+describe('NoteEditRoute', () => {
+  it('NoteEditRoute should render when location.state is not empty', () => {
+    renderNoteEditRoute(locationData, historyData);
+    screen.debug();
```
Comment (artem-blazhko) [inline src/routes/NoteEditRoute.test.js:65]: Could you remove it?

File: src/routes/NoteEditRoute.test.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  'listen': () => {},
+
+
+};
+const match = { params: { noteId: 'editNoteRouteID' },
+  path: 'editPath',
+  url: '{{ env.FOLIO_MD_REGISTRY }}/_/proxy/modules' };
+const locationData = historyData.location;
+
+const renderNoteEditRoute = (data1, data2) => render(
```
Comment (artem-blazhko) [inline src/routes/NoteEditRoute.test.js:57]: Could you use logical names for function arguments?

File: src/routes/NoteEditRoute.test.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  'createHref': () => {},
+  'push': () => {},
+  'replace': () => {},
+  'go': () => {},
+  'goBack': () => {},
+  'goForward': () => {},
+  'block': () => {},
+  'listen': () => {},
+
+
```
Comment (artem-blazhko) [inline src/routes/NoteEditRoute.test.js:50]: Could you remove empty spaces?

File: src/routes/NoteEditRoute.test.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    }
+  },
+  'createHref': () => {},
+  'push': () => {},
+  'replace': () => {},
+  'go': () => {},
+  'goBack': () => {},
+  'goForward': () => {},
+  'block': () => {},
+  'listen': () => {},
```
Comment (artem-blazhko) [inline src/routes/NoteEditRoute.test.js:48]: Could you remove quotes around object properties?

Comment (artem-blazhko) [reply src/routes/NoteEditRoute.test.js:48]: Could you move that mock data to some shared place?
I see that you use the same object here - https://github.com/folio-org/ui-requests/pull/1039/files

File: src/routes/NoteEditRoute.test.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  });
+  it('Request page should render when location.state is empty', async () => {
+    const historyProps = {
+      ...historyData,
+      'location': {
+        'state': '',
+        'hash': '',
+        'key': '9pb09t',
+        'pathname': '/users/notes/new',
+        'search': ''
```
Comment (artem-blazhko) [inline src/routes/NoteEditRoute.test.js:78]: Could you remove quotes around object properties?

File: src/routes/NoteEditRoute.test.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    <NoteEditRoute location={data1} history={data2} match={match} />
+  </Router>
+);
+describe('NoteEditRoute', () => {
+  it('NoteEditRoute should render when location.state is not empty', () => {
+    renderNoteEditRoute(locationData, historyData);
+    screen.debug();
+    userEvent.click(screen.getByRole('button', { name: 'renderReferredRecord' }));
+
+    expect(screen.getByText('NoteEditRoute'));
```
Comment (artem-blazhko) [inline src/routes/NoteEditRoute.test.js:68]: What is the expectation of this test?

File: src/routes/NoteEditRoute.test.js [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  });
+  it('Request page should render when location.state is empty', async () => {
```
Comment (artem-blazhko) [inline src/routes/NoteEditRoute.test.js:70]: I think you have a mistake here. Did you mean "should **not**"?
Do you really need to have **async** function here?

**Review by artem-blazhko — CHANGES_REQUESTED** [review]
Could you add appropriate changes to CHANGELOG.md file?
Could you add test that checks redirect functionality?

File: src/routes/NoteEditRoute.test.js (hunk 7 above)
Comment (cchitneedi) [reply src/routes/NoteEditRoute.test.js]: Changes done as suggested

File: CHANGELOG.md [hunk 8]
```diff
@@ -19,7 +19,8 @@
 * Cannot save item request from item with status On Order, because need a barcode. Refs UIREQ-924.
 * Cover `src/components/PrintButton/PrintButton.js` file by RTL/Jest tests. Refs UIREQ-886.
 * Cover `src/components/Loading/Loading.js` file by RTL/Jest tests. Refs UIREQ-885.
-* Leverage cookie-based authentication in all API requests. Refs UIREQ-861.
+* Create Jest/RTL test for NoteCreateRoute.js Refs: UIREQ-943.
+* Create Jest/RTL test for NoteCreateRoute.js Refs: UIREQ-944.
```
Comment (artem-blazhko) [inline CHANGELOG.md:23]: Why did you remove `Leverage cookie-based authentication in all API requests. Refs UIREQ-861.`?

Also, I suppose that `Create Jest/RTL test for NoteCreateRoute.js Refs: UIREQ-943.` is redundant.

File: test/jest/__mock__/stripesSmartComponents.mock.js [hunk 9]
```diff
@@ -4,6 +4,22 @@ jest.mock('@folio/stripes/smart-components', () => ({
...
   makeQueryFunction: jest.fn((value) => value),
   CheckboxFilter: jest.fn(() => null),
+  NoteCreatePage: (props) => {
+    return (
+      <div>
+        <div>NoteCreatePage</div>
+        {props.renderReferredRecord()}
+      </div>
+    );
+  },
```
Comment (artem-blazhko) [inline test/jest/__mock__/stripesSmartComponents.mock.js:14]: I do not see usage of `NoteCreatePage` inside `NoteEditRoute`. Could you remove it?

**Review by artem-blazhko — CHANGES_REQUESTED** [review]

File: test/jest/__mock__/stripesSmartComponents.mock.js (hunk 9 above)
Comment (cchitneedi) [reply test/jest/__mock__/stripesSmartComponents.mock.js]: Updated file

File: src/routes/NoteEditRoute.test.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+  it('Request page should render when location.state is empty', () => {
+    const historyProp = {
+      ...historyData,
+      location : {
+        hash: '',
+        key: '9pb09t',
+        pathname: '/users/notes/new',
+        search: '',
+        state: ''
+      },
```
Comment (artem-blazhko) [inline src/routes/NoteEditRoute.test.js:37]: Looks like some properties are not necessary here

```suggestion
      location : {
        state: ''
      },
```

Comment (cchitneedi) [reply src/routes/NoteEditRoute.test.js:37]: Hi @artem-blazhko , the following properties are added in order to avoid console warnings. Below is the attached screenshot for reference:
image

**Review by artem-blazhko — APPROVED** [review]

**Review by Dmitriy-Litvinenko — APPROVED** [review]
Hello. Thank you. Could you please update or rebase branch and merge it?
